substrate/client/network/src/litep2p/discovery.rs

```diff
@@ -72,11 +73,9 @@ const GET_RECORD_REDUNDANCY_FACTOR: usize = 4;
 /// The maximum number of tracked external addresses we allow.
 const MAX_EXTERNAL_ADDRESSES: u32 = 32;
 
-/// Minimum number of confirmations received before an address is verified.
-///
-/// Note: all addresses are confirmed by libp2p on the first encounter. This aims to make
-/// addresses a bit more robust.
-const MIN_ADDRESS_CONFIRMATIONS: usize = 2;
+/// Number of times observed address is received from different peers before it is confirmed as
+/// external.
+const MIN_ADDRESS_CONFIRMATIONS: usize = 3;
```

Comment:
We are now waiting for 3 confirmations in an LRU of 32 addresses. Are we losing precision somehow with this setup? Should we bump `MAX_EXTERNAL_ADDRESSES` to a higher value, or are we able to discover addresses just like before?

Reply:
There was a bug, and when we received the first address we inserted an empty hash map, and only after that counted confirmations. So it was always 3 confirmations, just the information about the first peer ID we got the address from was lost.